kickstarter/ios-oss
# [NT-1055] LandingPage deeplink handling bug fix

# 📲 What

Fixes a bug where a LandingPage is presented and immediately dismissed if the app is launched through deeplink.

# 🤔 Why

It's a bug and we don't like bugs.

# 🛠 How

- This was caused by a race condition between the `goToLandingPage` and the `deeplinkActivated`. So in order to solve this, a `take(until:)` function was added to prevent the `deeplinkActivated` from emitting if `goToLandingPage` already emitted.


# 👀 See

| Before 🐛 | After 🦋 |
| --- | --- |
| ![deeplink_bug](https://user-images.githubusercontent.com/3709676/78594793-e8e29b00-7816-11ea-80e4-2d93d38681e8.gif) | ![deeplink_fix](https://user-images.githubusercontent.com/3709676/78594802-eed87c00-7816-11ea-8bf7-a09ff90769b2.gif) |

# ✅ Acceptance criteria

In order to test this, follow the steps:
1. Set Launch setting to `Wait for executable to be launched`
2. On line `1116`, return `true` immediately after the open bracket:
   ```
   private func shouldGoToLandingPage() -> Bool { return true
```
3. Add a 2 sec delay to the `openUrl` property:
```
  let openUrl = self.applicationOpenUrlProperty.signal
      .delay(2, on: AppEnvironment.current.scheduler)
      .skipNil()
```
4. Build the app on a device in the release configuration
5. Click the link to open the app through deeplink. I tested using [this link]( https://www.kickstarter.com/projects/chelsea-punk/chelsea-punk-band-the-final-album))

- [ ] The landing page should be presented on the first time.
- [ ] Close the app and tap the link again. The app should show the Project Page and the Landing Page should **not** be presented.
